### PR TITLE
Added scala native

### PR DESCRIPTION
**Pull request by delneg**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -260,6 +260,31 @@ jobs:
...
+      - name: Build
+        shell: cmd
+        run: |
+          cd scala
+          sbt run
+          sbt --version
+          (
+          echo LANGUAGE=Scala
+          echo COMPILER=Scala 3.2.2 Scala-native 0.4.10
+          )>info.txt
```
Comment (MichalStrehovsky) [inline .github/workflows/ci.yml:279]: We need to put this in scala/target/scala-3.2.2 or GitHub will not properly pack it and and the reporting script won't work (see the artifacts from the workflow run summary).

File: scala/build.sbt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+// set to Debug for compilation details (Info is default)
+logLevel := Level.Info
+
+// import to add Scala Native options
+import scala.scalanative.build._
+
+// defaults set with common options shown
+nativeConfig ~= { c =>
+  c.withLTO(LTO.full) // thin
```
Comment (MichalStrehovsky) [inline scala/build.sbt:13]: We should not enable LTO. C/C++/others don't enable LTO. Only whatever the release build enables by default

File: scala/build.sbt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    .withMode(Mode.releaseSize) // releaseFast
+    .withGC(GC.immix) // commix
```
Comment (MichalStrehovsky) [inline scala/build.sbt:15]: Leave GC at default.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (delneg) [reply .github/workflows/ci.yml]: Done

File: scala/build.sbt [hunk 4]
```diff
@@ -10,7 +10,7 @@ import scala.scalanative.build._
 
 // defaults set with common options shown
 nativeConfig ~= { c =>
-  c.withLTO(LTO.full) // thin
+  c.withLTO(LTO.none) // thin
```
Comment (MichalStrehovsky) [inline scala/build.sbt:13]: We can't just skip these lines to leave things at defaults?

Comment (delneg) [reply scala/build.sbt:13]: Well, the project was generated using `sbt new scala-native/scala-native.g8` as per docs (https://scala-native.org/en/stable/user/sbt.html)
I didn't add these lines myself, and it seems to work fine without them but we at least need "Release" mode
I.e.
```
// defaults set with common options shown
nativeConfig ~= { c =>
  c.withMode(Mode.release)
}```

**Comment by MichalStrehovsky** [comment]
Thank you!

File: scala/build.sbt (hunk 4 above)
Comment (MichalStrehovsky) [reply scala/build.sbt]: I made the wrong call on this. If this file was generated with `sbt new scala-native/scala-native.g8`, these settings are the language defaults for native compilation (i.e. it was decided that the expected behavior is to have slow builds, but better perf - a tradeoff others don't do, but that's fine). Sizegame should be measuring the experience of an average developer and average developer is not going to go and edit these unless there's a need. I'll make a PR to put back the original.

**Pull request merged**